libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -832,6 +832,29 @@ bool ExpressionCompiler::visit(FunctionCall const& _functionCall)
 				StorageByteArrayElement(m_context).storeValue(*type, _functionCall.location(), true);
 			break;
 		}
+		case FunctionType::Kind::ByteArrayPop:
+		case FunctionType::Kind::ArrayPop:
+		{
+			_functionCall.expression().accept(*this);
+			solAssert(function.parameterTypes().size() == 0, "");
+
+			ArrayType const& arrayType = dynamic_cast<ArrayType const&>(
+				*dynamic_cast<MemberAccess const&>(_functionCall.expression()).expression().annotation().type);
+			
+			solAssert(arrayType.dataStoredIn(DataLocation::Storage), "");
+
+			// stack: ArrayReference
```

Comment:
I'm wondering whether we should also just move these into `ArrayUtils::decrementDynamicArraySize` (and then call it `popStorageArrayElement` or something).

Reply:
That's a great idea. We should try to gradually pull code out of this function.